### DISCUSSION TITLE
Installation Doubt

### DISCUSSION THREAD

**Discussion by TheLastSilva** [snipem/gt7dashboard#24] [Q&A] [answered]
Congratulations on the work, I'm very happy to be able to say this to you.

I'm having difficulty installing the dashboard, as I don't know much about programming I'm here asking for help, to understand what I need to do.  When I type "http://localhost:5006/gt7dashboard-main" the page appears blank.

Follows prompt message:

Collecting scipy~=1.9.0 (from -r requirements.txt (line 5))
  Using cached scipy-1.9.3.tar.gz (42.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [37 lines of output]
      + meson setup C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a\.mesonpy-wjjymu86 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a\.mesonpy-wjjymu86\meson-python-native-file.ini
      The Meson build system
      Version: 1.3.2
      Source dir: C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a
      Build dir: C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a\.mesonpy-wjjymu86
      Build type: native build
      Project name: SciPy
      Project version: 1.9.3
      Activating VS 17.8.6
      C compiler for the host machine: cl (msvc 19.38.33135 "Microsoft (R) C/C++ Optimizing Compiler VersÆo 19.38.33135 para x64")
      C linker for the host machine: link link 14.38.33135.0
      C++ compiler for the host machine: cl (msvc 19.38.33135 "Microsoft (R) C/C++ Optimizing Compiler VersÆo 19.38.33135 para x64")
      C++ linker for the host machine: link link 14.38.33135.0
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Compiler for C supports arguments -Wno-unused-but-set-variable: NO
      Compiler for C supports arguments -Wno-unused-but-set-variable: NO (cached)
      Compiler for C supports arguments -Wno-unused-function: NO
      Compiler for C supports arguments -Wno-conversion: NO
      Compiler for C supports arguments -Wno-misleading-indentation: NO
      Compiler for C supports arguments -Wno-incompatible-pointer-types: NO
      Library m found: NO
..\meson.build:57:0: ERROR: Unknown compiler(s): [['ifort'], ['gfortran'], ['flang'], ['pgfortran'], ['g95']]
      The following exception(s) were encountered:
      Running `ifort --version` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `ifort -V` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `gfortran --version` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `gfortran -V` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `flang --version` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `flang -V` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `pgfortran --version` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `pgfortran -V` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `g95 --version` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"
      Running `g95 -V` gave "[WinError 2] O sistema não pode encontrar o arquivo especificado"

      A full log can be found at C:\Users\aason\AppData\Local\Temp\pip-install-6wsr446d\scipy_30748d1f24574887968c350bf8b25d5a\.mesonpy-wjjymu86\meson-logs\meson-log.txt
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
2024-02-14 11:15:57,925 Starting Bokeh server version 3.1.1 (running on Tornado 6.4)
2024-02-14 11:16:02,423 User authentication hooks NOT provided (default user enabled)
2024-02-14 11:16:02,455 Bokeh app running at: http://localhost:5006/gt7dashboard-main
2024-02-14 11:16:02,455 Starting Bokeh server with process id: 9904
2024-02-14 11:16:56,603 Error running application handler <bokeh.application.handlers.directory.DirectoryHandler object at 0x0000019A63E7BE30>: No module named 'imp'
File 'salsa20.py', line 11, in <module>:
import imp Traceback (most recent call last):
  File "C:\Users\aason\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\bokeh\application\handlers\code_runner.py", line 231, in run
    exec(self._code, module.__dict__)
  File "C:\gt7dashboard-main\main.py", line 24, in <module>
    from gt7dashboard import gt7communication, gt7diagrams, gt7help, gt7helper, gt7lap
  File "C:\gt7dashboard-main\gt7dashboard\gt7communication.py", line 12, in <module>
    from salsa20 import Salsa20_xor
  File "C:\Users\aason\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\salsa20.py", line 11, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'

2024-02-14 11:16:57,613 WebSocket connection opened
2024-02-14 11:16:57,614 ServerConnection created

**Comment by snipem** [comment, marked as answer]
I think your python is too new … only python 3.11 is supported yet because of the salsa20 dependency

**Reply by TheLastSilva** [reply]
It's true, I had installed Python 3.12, with your answer I installed Python 3.11, and the dashboard appeared.
I'm running tests to learn how to use it. Since today my connection is unstable, I'm having a hard time. Thanks for the reply.
